lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -33,20 +70,183 @@ def onCreate(self, *args):
+        allKinds = Kind.Registry.GetAllKinds()
+        allKinds.remove(Kind.Tokens.model)
+        knownKinds = [Kind.Tokens.group, Kind.Tokens.assembly, Kind.Tokens.component, Kind.Tokens.subcomponent]
+        temp1 = [ele for ele in allKinds if ele not in knownKinds]
+        knownKinds.extend(temp1)
+
+        # If this prim's kind is not registered, we need to manually
+        # add it to the list.
+        model = Usd.ModelAPI(self.prim)
+        primKind = model.GetKind()
+        if primKind not in knownKinds:
+            knownKinds.insert(0, primKind)
+        if '' not in knownKinds:
+            knownKinds.insert(0, '')    # Set metadata value to "" (or empty).
+
+        self.kind = cmds.optionMenuGrp(label='Kind',
+                                       cc=self._onKindChanged,
+                                       ann='Kind is a type of metadata (a pre-loaded string value) used to classify prims in USD. Set the classification value from the dropdown to assign a kind category to a prim. Set a kind value to activate selection by kind.')
+
+        for ele in knownKinds:
+            cmds.menuItem(label=ele)
+
+        # Metadata: Active
+        self.active = cmds.checkBoxGrp(label='Active',
+                                       ncb=1,
+                                       cc1=self._onActiveChanged,
+                                       ann='If selected, the prim is set to active and contributes to the composition of a stage. If a prim is set to inactive, it doesn’t contribute to the composition of a stage (it gets striked out in the Outliner and is deactivated from the Viewport).')
+
+        # Metadata: Instanceable
+        self.instan = cmds.checkBoxGrp(label='Instanceable',
+                                       ncb=1,
+                                       cc1=self._onInstanceableChanged,
+                                       ann='If selected, instanceable is set to true for the prim and the prim is considered a candidate for instancing. If deselected, instanceable is set to false.')
+
+        # Get all the other Metadata and remove the ones above, as well as a few
+        # we don't ever want to show.
+        allMetadata = self.prim.GetAllMetadata()
+        keysToDelete = ['kind', 'active', 'instanceable', 'typeName', 'documentation']
+        for key in keysToDelete:
+            allMetadata.pop(key, None)
+        if allMetadata:
+            cmds.separator(h=10, style='single', hr=True)
+
+            for k in allMetadata:
+                # All extra metadata is for display purposes only - it is not editable, but we
+                # allow keyboard focus so you copy the value.
+                mdLabel = getPrettyName(k) if useNiceName else k
+                self.extraMetadata[k] = cmds.textFieldGrp(label=mdLabel, editable=False, enableKeyboardFocus=True)
+
+        # Update all metadata values.
+        self.refresh()
+
+    def refresh(self):
```

Comment:
Called when initially creating the UI to populate fields with value and also when a USD notif is received that has changed fields.